backend/src/Controllers/finalizeElectionController.ts

```diff
@@ -0,0 +1,56 @@
+import ServiceLocator from '../ServiceLocator';
+import Logger from '../Services/Logging/Logger';
+import { permissions } from '../../../domain_model/permissions';
+import { expectPermission } from "./controllerUtils";
+import { BadRequest, InternalServerError } from "@curveball/http-errors";
+import { Invites } from "../Services/Email/EmailTemplates"
+
+var ElectionsModel = ServiceLocator.electionsDb();
+var ElectionRollModel = ServiceLocator.electionRollDb();
+var EmailService = ServiceLocator.emailService();
+
+const className = "election.Controllers";
+
+const finalizeElection = async (req: any, res: any, next: any) => {
+    Logger.info(req, `${className}.finalize ${req.election.election_id}`);
+    expectPermission(req.user_auth.roles, permissions.canEditElectionState)
+    if (req.election.state !== 'draft') {
+        var msg = "Election already finalized";
+        Logger.info(req, msg);
+        throw new BadRequest(msg)
+    }
+    var failMsg = "Failed to update Election";
+    req.election.state = 'finalized'
+    const updatedElection = await ElectionsModel.updateElection(req.election, req, `Finalizing election`);
+    if (!updatedElection) {
+        Logger.info(req, failMsg);
+        throw new BadRequest(failMsg)
+    }
+    req.election = updatedElection
```

Comment:
what is the need to store the election back in the request object?  I would avoid this whenever possible as it becomes unclear where this data is used.

Reply:
Fixed, was from old code.